plugins/module_utils/network/sonic/config/drop_counter/drop_counter.py

```diff
@@ -0,0 +1,408 @@
+
+    def get_delete_drop_counter_requests(self, commands, is_delete_all):
+        """This method returns a list of delete requests generated from commands"""
+        requests = []
+
+        if not commands:
+            return requests
+        if is_delete_all:
+            requests.append(self.get_delete_drop_counter_request())
+            return requests
+
+        for counter in commands:
+            name = counter.get('name')
+            if len(counter) == 1:
+                requests.append(self.get_delete_drop_counter_request(name))
+                continue
+
+            alias = counter.get('alias')
+            counter_description = counter.get('counter_description')
+            counter_type = counter.get('counter_type')
+            enable = counter.get('enable')
+            group = counter.get('group')
+            mirror = counter.get('mirror')
+            reasons = counter.get('reasons')
+
+            if alias:
+                requests.append(self.get_delete_drop_counter_request(name, 'alias'))
+            if counter_description:
+                requests.append(self.get_delete_drop_counter_request(name, 'desc'))
+            if counter_type:
+                requests.append(self.get_delete_drop_counter_request(name, 'type'))
+            if enable is not None:
+                requests.append(self.get_delete_drop_counter_request(name, 'status'))
+            if group:
+                requests.append(self.get_delete_drop_counter_request(name, 'group'))
+            if mirror:
+                requests.append(self.get_delete_drop_counter_request(name, 'mirror'))
+            if reasons:
+                for reason in reasons:
+                    attr = 'reasons=' + reason
+                    requests.append(self.get_delete_drop_counter_request(name, attr))
+
+        return requests
+
+    @staticmethod
+    def get_delete_drop_counter_request(name=None, attr=None):
+        """This method formulates the URL and returns a delete request"""
+        url = DROP_COUNTER_PATH
+        if name:
+            url += '/DEBUG_COUNTER/DEBUG_COUNTER_LIST=%s' % (name)
+        if attr:
+            url += '/%s' % (attr)
+        return {'path': url, 'method': DELETE}
+
+    def get_replaced_config(self, want, have):
+        """This method returns the drop counter configuration to be deleted and the respective delete requests"""
+        config_list, requests = [], []
+        self.sort_lists_in_config(want)
+        self.sort_lists_in_config(have)
+
+        if not want or not have:
+            return config_list, requests
+
+        counter_dict = {counter.get('name'): counter for counter in have}
+        for counter in want:
+            name = counter.get('name')
+            cfg_counter = counter_dict.get(name)
+
+            if not cfg_counter:
+                continue
+            if counter != cfg_counter:
+                config_list.append(cfg_counter)
+
+        self.remove_default_entries(config_list)
```

Comment:
Since the command contains entire 'have' config of counter, removing default will only remove the default resulting in the deletion of other options even if the configs are equal (have = want + default).

Please remove the defaults in both 'have' and 'want' (copy of both) before the equality check to ensure delete command is not generated for equal configs.